packages/backend/src/vm-manager.ts

```diff
@@ -0,0 +1,200 @@
+    }
+
+    try {
+      const command = this.generateLaunchCommand(diskImage);
+
+      if (command.length === 0) {
+        throw new Error(
+          'Unable to generate the launch command for the VM, ensure you are on the appropriate OS and architecture.',
+        );
+      }
+
+      await extensionApi.process.exec('sh', ['-c', `${command.join(' ')}`]);
+    } catch (e) {
+      this.handleError(e);
+    }
+  }
+
+  // We only support running one VM at at a time, so we kill the process by reading the pid from the universal pid file we use.
+  public async stopCurrentVM(): Promise<void> {
+    try {
+      await extensionApi.process.exec('sh', ['-c', `kill -9 \`cat ${pidFile}\``]);
+    } catch (e) {
+      // Ignore if it contains 'No such process' as that means the VM is already stopped / not running.
+      if (e instanceof Error && 'stderr' in e && typeof e.stderr === 'string' && e.stderr.includes('No such process')) {
+        return;
+      }
+      this.handleError(e);
+    }
+  }
+
+  // Prerequisite checks before launching the VM which includes checking if QEMU is installed as well as other OS specific checks.
+  public async checkVMLaunchPrereqs(): Promise<string | undefined> {
+    const diskImage = this.getDiskImagePath();
+    if (!fs.existsSync(diskImage)) {
+      return `Raw disk image not found at ${diskImage}. Please build a .raw disk image first.`;
+    }
+
+    if (!this.isArchitectureSupported()) {
+      return `Unsupported architecture: ${this.build.arch}`;
+    }
+
+    // Future support for Mac Intel, Linux ARM, Linux X86 and Windows ARM, Windows X86 to be added here.
+    if (isMac() && isArm()) {
+      return this.checkMacPrereqs();
+    } else {
+      return 'Unsupported OS. Only MacOS Silicon is supported.';
+    }
+  }
+
+  private getDiskImagePath(): string {
+    return path.join(this.build.folder, rawImageLocation);
+  }
+
+  private isArchitectureSupported(): boolean {
+    return this.build.arch === 'amd64' || this.build.arch === 'arm64';
+  }
+
+  private checkMacPrereqs(): string | undefined {
+    const installDisclaimer = 'Please install qemu via our installation document';
+    if (!fs.existsSync(macQemuX86Binary)) {
+      return `QEMU x86 binary not found at ${macQemuX86Binary}. ${installDisclaimer}`;
+    }
+    if (this.build.arch === 'arm64' && !fs.existsSync(macQemuArm64Binary)) {
+      return `QEMU arm64 binary not found at ${macQemuArm64Binary}. ${installDisclaimer}`;
+    }
+    if (this.build.arch === 'arm64' && !fs.existsSync(macQemuArm64Edk2)) {
+      return `QEMU arm64 edk2-aarch64-code.fd file not found at ${macQemuArm64Edk2}. ${installDisclaimer}`;
+    }
+    return undefined;
+  }
+
+  // Supported: MacOS Silicon
+  // Unsupported: MacOS Intel, Linux, Windows
+  private generateLaunchCommand(diskImage: string): string[] {
+    // Future support for Mac Intel, Linux ARM, Linux X86 and Windows ARM, Windows X86 to be added here.
+    if (isMac() && isArm()) {
```

Comment:
if we had proper method in macArm implementation, we would not need the is and is